x/ibc/testing/solomachine.go

```diff
@@ -53,7 +53,7 @@ func (solo *Solomachine) ClientState() *solomachinetypes.ClientState {
 
 // ConsensusState returns a new solo machine ConsensusState instance
 func (solo *Solomachine) ConsensusState() *solomachinetypes.ConsensusState {
-	publicKey, err := std.DefaultPublicKeyCodec{}.Encode(solo.PublicKey)
+	publicKey, err := tx.PubKeyToAny(solo.PublicKey)
```

Comment:
so the issue is the tests currently rely on being able to change the value of `solo.PublicKey` after this occurs, but packing to any uses a pointer whose underlying value might be changed. Does that make sense?

there's a two ways to fix this:
- make a copy of the public key and pack that to an any
- adjust how to the tests interact to avoid changing this value

I can start trying to fix with the latter solution because I'm not sure how to copy the public key

Reply:
I'm running into this issue now:
```
	panic: ConsensusState PublicKey is not crypto.PubKey
```